tests/Functional/Platform/AlterUuidColumnTest.php

```diff
@@ -0,0 +1,150 @@
+    }
+
+    public function testAlterUuidInForeignRelation(): void
+    {
+        $sm = $this->connection->createSchemaManager();
+
+        $parentTable = static::prepareLegacyUuidTable('parent_uuid_table', 'id');
+        $parentTable->setPrimaryKey(['id']);
+        $this->dropAndCreateTable($parentTable);
+
+        $this->connection->executeStatement(
+            sprintf(
+                'INSERT INTO parent_uuid_table VALUES ("%s")',
+                static::DEFAULT_UUID4,
+            ),
+        );
+
+        $childTable = static::prepareLegacyUuidTable('child_uuid_table', 'parent_uuid_id');
+        $childTable->addColumn('id', Types::INTEGER)->setAutoincrement(true);
+        $childTable->setPrimaryKey(['id']);
+
+        $childTable = static::prepareForeignKey($childTable, 'parent_uuid');
+        $this->dropAndCreateTable($childTable);
+
+        $this->connection->executeStatement(
+            sprintf(
+                'INSERT INTO child_uuid_table (`parent_uuid_id`) VALUES ("%s")',
+                static::DEFAULT_UUID4,
+            ),
+        );
+
+        $childTable->removeForeignKey('fk_parent_uuid_id');
+        $diff = $sm->createComparator()
+            ->compareTables($sm->introspectTable('child_uuid_table'), $childTable);
+
+        $sm->alterTable($diff);
+
+        $parentTable->getColumn('id')
+            ->setType(Type::getType(Types::GUID));
+
+        $diff = $sm->createComparator()
+            ->compareTables($sm->introspectTable('parent_uuid_table'), $parentTable);
+
+        $sm->alterTable($diff);
+
+        $childTable->getColumn('parent_uuid_id')
+            ->setType(Type::getType(Types::GUID));
+
+        $childTable = static::prepareForeignKey($childTable, 'parent_uuid');
+
+        $diff = $sm->createComparator()
+            ->compareTables($sm->introspectTable('child_uuid_table'), $childTable);
+
+        $sm->alterTable($diff);
+
+        self::assertInstanceOf(GuidType::class, $parentTable->getColumn('id')->getType());
+        self::assertInstanceOf(GuidType::class, $childTable->getColumn('parent_uuid_id')->getType());
+
+        // Verify data integrity
+        $resultUuid = $this->connection
+            ->executeQuery('SELECT id from parent_uuid_table')
+            ->fetchOne();
+
+        static::assertEquals(static::DEFAULT_UUID4, $resultUuid);
+
+        $resultUuid = $this->connection
+            ->executeQuery('SELECT parent_uuid_id from child_uuid_table')
+            ->fetchOne();
+
+        static::assertEquals(static::DEFAULT_UUID4, $resultUuid);
+    }
+
+    /**
+     * Create legacy Uuid table.
+     */
+    protected function prepareLegacyUuidTable(string $tableName, string $uuidField): Table
+    {
+        $table = new Table($tableName);
+        $table->addColumn($uuidField, Types::STRING, [
+            'length'  => 36,
+            'notnull' => false,
+            'fixed'   => true,
+        ]);
+
+        return $table;
```

Comment:
Don't use the schema manager to create this table. Use plain SQL to create the table as DBAL 4.0 would've created it if we configured a GUID column.

Reply:
Done. Now the table is created using plain SQL and then the table is introspected.